### DISCUSSION TITLE
Nomenclature Discussion for Tensor-LEED Parameters

### DISCUSSION THREAD

**Discussion by alexmiame** [viperleed/viperleed#275] [Ideas]
Edit: Updated the TL_Nomenclature PDF with suggestions.

Hi everyone!

With the new tensor-LEED implementation, we’ve been revisiting formal definitions and nomenclature for the parameters subject to optimization. Specifically, we’ve been debating how to name the sets over which we enumerate these parameters.

For tensor-LEED calculations, we need to calculate perturbations for every possible chemical species at every geometrical position in the slab. In ViPErLEED, we currently use the `Atom` and `Sitetype `classes for these enumerations. However, the naming can sometimes be ambiguous—for instance, our `Atom` class represents sets of all real-world atoms that might occupy a site. While this isn’t a critical issue as long as we’re consistent and clear, it may still be worth rethinking some names.

For the new implementation (and the associated paper), we need to be more explicit in describing the details of each calculation. @michele-riva , @Paulhai7, and I have been brainstorming potential names and definitions.

I’ve attached a PDF with an overview of the symbols, names, and descriptions. We’d love to hear your thoughts or alternative suggestions. We’re open to anything that makes sense! For now, we’ll start using these names in the new implementation, though if we decide to rename things, we may update `viperleed.calc` accordingly.

[TL_Nomenclature.pdf](https://github.com/user-attachments/files/17874361/TL_Nomenclature.pdf)

**Comment by alexmiame** [comment]
It would be good to hear some opinions by @fkraushofer and @schmid-iap in particular.

**Comment by alexmiame** [comment]
One question we had, was if we should call the optimization vector in tensor-LEED $x$, or if we should call it $\delta x$ and reserve $x$ for optimizations including full-dynamic calculations.

**Comment by michele-riva** [comment]
Perhaps a clarification on what would change in the current `viperleed.calc`: we would introduce a new `Site` class (or whatever it will be called) that replaces the current `Atom` class. The new `Atom` class would be a real atom: multiple `Atom`s can occupy the same `Site`.

The thing I'm not yet convinced about for what concerns the naming is that parts of code like `for atom in slab` would need to be replaced with `for site in slab`. I find the former (current) to be clearer than the latter.

**Comment by fkraushofer** [comment]
Overall, I think this makes sense; the only issue with "site" I have is that "sitetype" might then be a bit confusing. On the other hand, it's perfectly descriptive, so maybe I just find it counterintuitive because I've personally been calling the sitetypes "sites" in my head for years.

> The thing I'm not yet convinced about for what concerns the naming is that parts of code like `for atom in slab` would need to be replaced with `for site in slab`. I find the former (current) to be clearer than the latter.

Hmm, it's true that "atom in slab" is more intuitive, but the drawback is that it's then easy to miss that this "atom" in many instances needs to be looped over again for the multiple elements it may contain, with possibly different properties. I think in this way "site" is more clear about what's actually going on in the code. I'll have to learn to rethink the nomenclature, but in principle my first impulse is that I approve of the change. It's certainly more descriptive that way.

> One question we had, was if we should call the optimization vector in tensor-LEED x , or if we should call it δ x and reserve x for optimizations including full-dynamic calculations.

I agree, **x** is rather confusing (I guess one can't technically "optimize x" in a full-dynamic optimization of the unit cell, only **a**, but still - it's an easy thing to mix up). Slightly better would be an upper-scale **X**, at least then there's a clear distinction, but this is also easily lost. Generally I think if your "general equation" for the optimization contains **x**, and someone only looks at that equation, it would be easy to interpret this incorrectly as "only optimizing positions". I would rather just pick some other letter that has no meaning yet. **c** for configuration is unfortunately also taken for the out-of-plane unit cell vector. **p** for parameter(set), perhaps? Then the p_i would be the single meta-parameters. **s** for "set" is also kind of intuitive, but then the s_i don't really make sense.

**Comment by alexmiame** [comment]
One other option I was thinking about was "Scatterer" or "Point Scatterer", as that is what we ultimately treat the (element, geometric site, site type) tuple as. However, then we'd have two things starting with "S" if we also go with Site...

For the optimization parameters, I see that $x$ can be a bit confusing. I'm fine with $p$. We could also use $\xi$ for the general ones and use some other letter for the "meta" parameters.

**Reply by michele-riva** [reply]
> One other option I was thinking about was "Scatterer" or "Point Scatterer", as that is what we ultimately treat the (element, geometric site, site type) tuple as. However, then we'd have two things starting with "S" if we also go with Site...

The other problem with `Scatterer` is that we thought of using the name for the `(Sitetype, Element)` pairs used for the `PHASESHIFTS`. Perhaps a more explicit name than `Site` would be `AtomicSite` (for which the same symbol _S_ would work)?

> For the optimization parameters, I see that _x_ can be a bit confusing. I'm fine with _p_. We could also use _ξ_ for the general ones and use some other letter for the "meta" parameters.

I fear _p_ is a bit overused, as we have plenty of other stuff that we call "parameter". We could use any other Greek letter: _χ_ (looks a bit like an _x_, though, which may be both an advantage and a disadvantage), _ζ_, or, as @amimre suggested, _ξ_ (in which case an alternative for the meta parameters could be _μ_ -- I'd avoid _m_ to prevent confusion with mass).

**Comment by alexmiame** [comment]
Ok, I've updated the PDF at the top based on the suggestions by @michele-riva and @fkraushofer.
I've used $\xi$ for the general parameters and $\mu$ for the meta parameters for now.
I've also added the "Scatterers" as (T,E) tuples for now, without a dedicated symbol.

I'll use these definitions for now, and we can revisit it once I have a first version of the manuscript.
